tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -237,4 +222,40 @@ object NativeConfig {
     }
   }
 
+  def predefinedLinktimeProperties(config: NativeConfig): LinktimeProperites = {
+    val linktimeInfo = "scala.scalanative.meta.linktimeinfo"
+    Map(
+      s"$linktimeInfo.isWindows" -> Platform.isWindows,
+      s"$linktimeInfo.isWeakReferenceSupported" -> {
+        config.gc == GC.Immix ||
+        config.gc == GC.Commix
+      }
+    )
+  }
+
+  def checkLinktimeProperties(properties: LinktimeProperites): Unit = {
+    def isNumberOrString(value: Any) = {
+      def hasSupportedType = value match {
+        case _: Boolean | _: Byte | _: Char | _: Short | _: Int | _: Long |
+            _: Float | _: Double | _: String =>
+          true
+        case _ => false
+      }
+
+      value != null && hasSupportedType
```

Comment:
`value != null` is redundant. If `value` is null, the `case _: T` will not match for any `T`, and therefore `case _ => false` will be used.